backend/db/migrate/20220406201351_add_launched_study_ref_to_launched_stage.rb

```diff
@@ -0,0 +1,19 @@
+class AddLaunchedStudyRefToLaunchedStage < ActiveRecord::Migration[6.1]
+  def change
+    reversible do |dir|
+      dir.up do
+        add_reference :launched_stages, :launched_study
+        execute 'update launched_stages lsa '\
+                'set launched_study_id = '\
+                '(select lsu.id from stages s join launched_studies lsu '\
+                'on s.study_id = lsu.study_id '\
+                'where s.id = lsa.stage_id and lsu.user_id = lsa.user_id)'
+        change_column_null :launched_stages, :launched_study_id, false
```

Comment:
the "ruby way" to do multi line strings is with HEREDOC: https://www.rubyguides.com/2018/11/ruby-heredoc/
```ruby
 execute <<-SQL
    update launched_stages lsa 
      set launched_study_id = (
        select lsu.id from stages s join launched_studies lsu on s.study_id = lsu.study_id 
                    where s.id = lsa.stage_id and lsu.user_id = lsa.user_id
      )
SQL
```

Also it doesn't matter too much since this is a one-off query, but I prefer to use join syntax vs sub select, and have started to dislike aliases because they seem to require more mental gymnastics to remember what table each alias refers to, and it's not _that_ much more work to type a few more chars 😁  Of course you have to use them for self joins but often they're not needed.  I think this could be:
```SQL
update launched_stages 
  set launched_study_id = launched_studies.id
from stages, launched_studies where
  stages.study_id = launched_stages.id
    and  stages.id = launched_stages.stage_id
    and launched_studies.user_id = launched_stages.user_id
```

Reply:
Yeah, my alias habit has developed from working at the DB console evolving the query by hand. :) The update w/ subselect is because I was lazy and had gotten the select to generate the correct list, first. I'll test yours, I too, like it better.